mm2src/coins/utxo/utxo_common.rs

```diff
@@ -680,17 +680,17 @@ impl<'a, T: AsRef<UtxoCoinFields> + UtxoTxGenerationOps> UtxoTxBuilder<'a, T> {
         };
 
         for utxo in self.available_inputs.clone() {
-            if self.update_fee_and_check_completeness(from.addr_format(), &actual_tx_fee) {
-                break;
-            }
-
             self.tx.inputs.push(UnsignedTransactionInput {
                 previous_output: utxo.outpoint,
                 prev_script: utxo.script,
                 sequence: SEQUENCE_FINAL,
                 amount: utxo.value,
             });
             self.sum_inputs += utxo.value;
+
+            if self.update_fee_and_check_completeness(from.addr_format(), &actual_tx_fee) {
+                break;
+            }
```

Comment:
> Should we leave it at the top and call it one last time after the loop instead?
> We should ideally also throw an error if the last completeness check (the one outside the loop) fails, that means we don't have enough inputs to cover outputs + fees.

This is much more complicated IMHO, I would rather leave it as is since it makes more sense for me. After adding an input we check if the transaction is complete or not (output value is covered by inputs), we discard all other inputs after that.

Reply:
Fixed by adding a check for `update_fee_and_check_completeness` that runs the for loop if it was false.